dispatches/renewables_case/battery/battery.py

```diff
@@ -0,0 +1,146 @@
+
+        self.initial_energy_throughput = Var(within=NonNegativeReals,
+                                             initialize=0.0,
+                                             doc="Cumulative energy throughput at t - 1",
+                                             units=pyunits.kWh)
+
+        self.state_of_charge = Var(self.flowsheet().config.time,
+                                   within=NonNegativeReals,
+                                   initialize=0.0,
+                                   doc="State of charge (energy), [0, self.nameplate_energy]",
+                                   units=pyunits.kWh)
+
+        self.elec_in = Var(self.flowsheet().config.time,
+                             within=NonNegativeReals,
+                             initialize=0.0,
+                             doc="Energy in",
+                             units=pyunits.kW)
+
+        self.elec_out = Var(self.flowsheet().config.time,
+                              within=NonNegativeReals,
+                              initialize=0.0,
+                              doc="Energy out",
+                              units=pyunits.kW)
+
+        self.energy_throughput = Var(self.flowsheet().config.time,
+                                     within=NonNegativeReals,
+                                     initialize=0.0,
+                                     doc="Cumulative energy throughput",
+                                     units=pyunits.kWh)
+
+        self.power_in = Port(noruleinit=True, doc="A port for electricity inflow")
+        self.power_in.add(self.elec_in, "electricity")
+
+        self.power_out = Port(noruleinit=True, doc="A port for electricity outflow")
+        self.power_out.add(self.elec_out, "electricity")
+
+        @self.Constraint(self.flowsheet().config.time)
+        def state_evolution(b, t):
+            if t == 0:
+                return b.state_of_charge[t] == b.initial_state_of_charge + (
+                        b.charging_eta * b.dt * b.elec_in[t]
+                        - b.dt / b.discharging_eta * b.elec_out[t])
+            return b.state_of_charge[t] == b.state_of_charge[t-1] + (
+                    b.charging_eta * b.dt * b.elec_in[t]
+                    - b.dt / b.discharging_eta * b.elec_out[t])
+
+        @self.Constraint(self.flowsheet().config.time)
+        def accumulate_energy_throughput(b, t):
+            if t == 0:
+                return b.energy_throughput[t] == b.initial_energy_throughput + b.dt * (b.elec_in[t] + b.elec_out[t]) / 2
+            return b.energy_throughput[t] == b.energy_throughput[t - 1] + b.dt * (b.elec_in[t] + b.elec_out[t]) / 2
+
+        @self.Constraint(self.flowsheet().config.time)
+        def state_of_charge_bounds(b, t):
+            return b.state_of_charge[t] <= b.nameplate_energy - b.degradation_rate * b.energy_throughput[t]
+
+        @self.Constraint(self.flowsheet().config.time)
+        def power_bound_in(b, t):
+            return b.elec_in[t] <= b.nameplate_power
```

Comment:
It depends on whether `nameplate_power` should be a Var or a Param. I had it as a Var since I thought it would be a variable in the conceptual design, but using its value as a bound would require it to be a fixed value. @jghouse88 thoughts?